app/controllers/api/vcs.php

```diff
@@ -673,39 +766,81 @@
-
-                    $runtimes = Config::getParam('runtimes');
-                    $runtimeDetail = \array_reverse(\array_filter(\array_keys($runtimes), function ($key) use ($runtime, $runtimes) {
-                        return $runtimes[$key]['key'] === $runtime;
-                    }))[0] ?? '';
-
-                    $repo['runtime'] = $runtimeDetail;
-                } catch (Throwable $error) {
-                    $repo['runtime'] = "";
-                    Console::warning("Runtime not detected for " . $repo['organization'] . "/" . $repo['name']);
+        $repos = batch(\array_map(function ($repo) use ($type, $github) {
+            return function () use ($repo, $type, $github) {
+                $files = $github->listRepositoryContents($repo['organization'], $repo['name'], '');
+                $files = \array_column($files, 'name');
+                $languages = $github->listRepositoryLanguages($repo['organization'], $repo['name']);
+                if ($type === 'framework') {
+                    try {
+                        $frameworkDetector = new Framework($files);
+                        $frameworkDetector
+                            ->addOption(new Flutter())
+                            ->addOption(new Nuxt())
+                            ->addOption(new Astro())
+                            ->addOption(new Remix())
+                            ->addOption(new SvelteKit())
+                            ->addOption(new NextJs());
+
+                        $detectedFramework = $frameworkDetector->detect();
+
+                        if ($detectedFramework) {
+                            $framework = $detectedFramework->getName();
+                        } else {
+                            $framework = 'other';
+                        }
+
+                        $frameworks = Config::getParam('frameworks');
+                        $frameworkDetail = \array_reverse(\array_filter(\array_keys($frameworks), function ($key) use ($framework, $frameworks) {
+                            return $frameworks[$key]['key'] === $framework;
+                        }))[0] ?? '';
+                        $repo['framework'] = $frameworkDetail;
+                    } catch (Throwable $error) {
+                        $repo['framework'] = "";
+                        Console::warning("Framework not detected for " . $repo['organization'] . "/" . $repo['name']);
+                    }
+                } else {
+                    try {
+                        $strategies = [
+                            new Strategy(Strategy::FILEMATCH),
+                            new Strategy(Strategy::LANGUAGES),
+                            new Strategy(Strategy::EXTENSION),
+                        ];
+
+                        foreach ($strategies as $strategy) {
+                            $runtimeDetector = new Runtime($strategy === Strategy::LANGUAGES ? $languages : $files, $strategy, 'npm');
```

Comment:
Let's not assume NPM here, do detection for package manager

Reply:
Packager is not required here since we are just detecting framework/ runtime and not any commands.